qa/L0_backend_python/examples/test.sh

```diff
@@ -38,10 +38,12 @@ rm -fr *.log python_backend/
 
 # Install torch
 pip3 uninstall -y torch
+pip3 uninstall -y numpy
+pip3 install "numpy>=2"
 if [ "$TEST_JETSON" == "0" ] && [[ ${TEST_WINDOWS} == 0 ]]; then
-    pip3 install torch==2.0.0+cu117 -f https://download.pytorch.org/whl/torch_stable.html torchvision==0.15.0+cu117
+    pip3 install torch==2.5.0 -f https://download.pytorch.org/whl/torch_stable.html torchvision==0.20.0
 else
-    pip3 install torch==2.0.0 -f https://download.pytorch.org/whl/torch_stable.html torchvision==0.15.0
+    pip3 install torch==2.5.0 -f https://download.pytorch.org/whl/torch_stable.html torchvision==0.20.0
```

Comment:
Note that the `if` condition installed a `+cu` version, and the `else` did not. In your change you have both if/else installing the same thing. Please either match the install pattern with the correct `+cuXXX` variant, or double check whether this if/else is required if not. CC @fpetrini15 who might know a bit more about the if/else split here.

Reply:
@rmccorm4 I think this is a code segment we discussed a few months back. It seems like since Jacky has fixed my original error here; https://github.com/triton-inference-server/server/commit/23a6c2189db716bc9d00e4ba346969be6ed69ba2. I think for non-Jetson Linux tests, we want to continue installing the cuda-enabled version. That said, these tests were passing fine even with my logical error from months ago, so it's unclear to me what we achieve by installing the cuda-enabled version.

Reply:
If this doesn't break windows/jetson, I'm happy to keep it. If it might, we could also try making the change only on the linux non-jetson block. ex:

```
if not jetson/windows:
  pip install numpy 2
  pip install torch 2.5
else: # if windows/jetson
  pip install numpy 1
  pip install torch 2.0
fi
``` 

Let me know if you have a preference for minimal blast radius @fpetrini15

Reply:
I think we should align with the original intention of the script and install the cuda-enabled version.

CC @Tabrizian if you have any insight into why we preferred it originally?

Reply:
I believe PyTorch with CUDA was not supported on Jetson. We also don't support GPU tensors on Jetson with Python backend so it could be related to that.